src/main/java/teammates/ui/webapi/GetFeedbackQuestionsAction.java

```diff
@@ -36,7 +36,11 @@ void checkSpecificAccessControl() throws UnauthorizedAccessException {
             break;
         case FULL_DETAIL:
             gateKeeper.verifyLoggedInUserPrivileges(userInfo);
-            gateKeeper.verifyAccessible(logic.getInstructorForGoogleId(courseId, userInfo.getId()), feedbackSession);
+            if (userInfo.isStudent) {
+                gateKeeper.verifyAccessible(getStudentOfCourseFromRequest(courseId), feedbackSession);
+            } else {
+                gateKeeper.verifyAccessible(logic.getInstructorForGoogleId(courseId, userInfo.getId()), feedbackSession);
+            }
             break;
```

Comment:
Could you elaborate on why is this change required? I don't think students need to see full detail.

Reply:
Have reverted the change after looking at it again. Instead of using `FULL_DETAIL`, I have made changes so that `STUDENT_RESULT` intent can be used.